packages/admin/src/pages/build-application/[applicationId]/[sectionId]/[questionId]/edit/question-content/getServerSideProps.test.tsx

```diff
@@ -0,0 +1,326 @@
+import { getApplicationFormSummary } from '../../../../../../../services/ApplicationService';
+import {
+  InferServiceMethodResponse,
+  Optional,
+  ValidationError,
+  getContext,
+  mockServiceMethod,
+} from 'gap-web-ui';
+import { parseBody } from '../../../../../../../utils/parseBody';
+import {
+  getQuestion,
+  patchQuestion,
+} from '../../../../../../../services/QuestionService';
+import ResponseTypeEnum from '../../../../../../../enums/ResponseType';
+
+jest.mock('../../../../../../../utils/parseBody');
+jest.mock('../../../../../../../services/ApplicationService');
+jest.mock('../../../../../../../services/QuestionService');
+
+const mockedGetApplicationFormSummary = jest.mocked(getApplicationFormSummary);
+const mockedGetQuestion = jest.mocked(getQuestion);
+const mockedPatchQuestion = jest.mocked(patchQuestion);
+
+const getDefaultAppFormSummary = (): InferServiceMethodResponse<
+  typeof getApplicationFormSummary
+> => ({
+  applicationName: 'Some application name',
+  applicationStatus: 'DRAFT',
+  audit: {
+    created: '2021-08-09T14:00:00.000Z',
+    lastPublished: '2021-08-09T14:00:00.000Z',
+    lastUpdatedBy: 'some-user',
+    lastUpdatedDate: '2021-08-09T14:00:00.000Z',
+    version: 1,
+  },
+  grantApplicationId: 'testApplicationId',
+  grantSchemeId: 'some-grant-scheme-id',
+  sections: [
+    {
+      sectionId: 'testSectionId',
+      sectionStatus: 'COMPLETE',
+      sectionTitle: 'some-section-title',
+      questions: [],
+    },
+  ],
+});
+
+const getDefaultQuestion = (): InferServiceMethodResponse<
+  typeof getQuestion
+> => ({
+  fieldTitle: 'Test Section Field Title',
+  hintText: 'Test hint text',
+  validation: { mandatory: 'true' },
+  responseType: ResponseTypeEnum.ShortAnswer,
+  questionId: 'testQuestionId',
+  adminSummary: '',
+  profileField: '',
+  fieldPrefix: '',
+  questionSuffix: '',
+  displayText: '',
+});
+
+const expectedGetRedirectObject = {
+  redirect: {
+    destination: `/service-error?serviceErrorProps={"errorInformation":"Something went wrong while trying to load this page.","linkAttributes":{"href":"/testResolvedURL","linkText":"Please return","linkInformation":" and try again."}}`,
+    statusCode: 302,
+  } as Redirect,
+};
+
+describe('getServerSideProps', () => {
+  beforeEach(() => {
+    mockServiceMethod(
+      mockedGetApplicationFormSummary,
+      getDefaultAppFormSummary
+    );
+
+    mockServiceMethod(mockedGetQuestion, getDefaultQuestion);
+  });
+
+  const getDefaultContext = (): Optional<GetServerSidePropsContext> => ({
+    params: {
+      applicationId: 'testApplicationId',
+      sectionId: 'testSectionId',
+      questionId: 'testQuestionId',
+    },
+    query: {
+      scrollPosition: '0',
+    },
+  });
+
+  it('Should return a back button href', async () => {
+    const value = await getServerSideProps(getContext(getDefaultContext));
+
+    if ('redirect' in value) throw new Error('Expected props');
```

Comment:
```suggestion
    expect(value.redirect).not.toExist();
```

Reply:
not sure if typescript is smart enough to update the types from this but will see if it complains

Reply:
This works but... I'd argue its much yuckier

```    const value = await getServerSideProps(getContext(getDefaultContext));

    expect(value).toEqual(
      expect.objectContaining({
        props: expect.objectContaining({
          pageData: expect.objectContaining({
            backButtonHref:
              '/build-application/testApplicationId/testSectionId',
          }),
        }),
      })
    );

Reply:
`Exclude<Awaited<ReturnType<T>>, NextRedirect>` does the magic